src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateDtoValidator.java

```diff
@@ -35,6 +39,9 @@
 @Component
 public class GermplasmUpdateDtoValidator {
 
+	private static final List<VariableType> ATTRIBUTE_TYPES =
+		Arrays.asList(VariableType.GERMPLASM_ATTRIBUTE, VariableType.GERMPLASM_PASSPORT);
```

Comment:
Maybe just reuse the `ALLOWED_ATTRIBUTE_TYPES` constant in AttributeValidator to reduce duplications?

Reply:
but in the context of my suggestion to have a Middleware call to expose a service to get all attribute and passport variables, this will be unnecessary

Reply:
I moved it to VariableType